rmc/pipeline/two_breaks/run_batches.py

```diff
@@ -210,147 +208,82 @@ def search_for_two_breaks(
-            # Also note that j needs to be checked here to ensure that j is also at the end of its loop
-            # (This check is necessary when transcripts have been split into multiple i, j windows
-            # across multiple rows)
-            (i == max_idx_i) & (j == max_idx_j),
-            (cur_max_chisq, cur_best_i, cur_best_j),
-            # If we haven't reached the end of the position list with index i,
-            # continue with the loop
-            hl.if_else(
-                j == max_idx_j,
-                # At end of j iteration, continue to next i index
-                # Set i to i + 1
-                # and set j to the larger value between i + 2 and start index value for j
-                # This is to avoid redundant work in larger transcripts; e.g.:
-                # start_idx_i = 0, start_idx_j = 50 ->
-                # using `hl.max()` here means that j will be reset to 50 rather than 2 on the second
-                # iteration of the loop will restart at 50
-                # Note that the j index should always be larger than the i index
-                loop_continue(
-                    i + 1,
-                    hl.min(hl.max(i + 2, start_idx_j), max_idx_j),
-                    start_idx_j,
-                    max_idx_i,
-                    max_idx_j,
-                    cur_max_chisq,
-                    cur_best_i,
-                    cur_best_j,
-                ),
-                # Otherwise, if j hasn't gotten to the maximum index,
-                # continue to the next j value for current i
-                loop_continue(
-                    i,
-                    j + 1,
-                    start_idx_j,
-                    max_idx_i,
-                    max_idx_j,
-                    cur_max_chisq,
-                    cur_best_i,
-                    cur_best_j,
+    # Create ArrayExpression of StructExpressions that stores each
+    # pair of positions (window breakpoints) and their corresponding chi square value
+    break_values_expr = hl.range(
+        group_ht.start_idx.i_start, group_ht.i_max_idx
+    ).flatmap(
+        lambda i: hl.range(group_ht.start_idx.j_start, group_ht.j_max_idx).map(
+            lambda j: hl.struct(
+                i=i,
+                j=j,
+                chisq=hl.nanmax(
+                    calculate_window_chisq(
+                        group_ht.max_idx,
+                        i,
+                        j,
+                        group_ht.cum_obs,
+                        group_ht.cum_exp,
+                        group_ht.total_oe,
+                        min_num_exp_mis,
+                    ),
+                    -1,
                 ),
-            ),
+            )
         )
-
+    )
+    # Filter ArrayExpression to only keep chi square values that are at least
+    # some minimum value (to shorten this array and save on storage)
+    break_values_expr.filter(lambda x: x.chisq >= min_chisq_threshold)
```

Comment:
```suggestion
    break_values_expr = break_values_expr.filter(lambda x: x.chisq >= min_chisq_threshold)
```
And same in `simultaneous_breaks.py`